Regional historical changes for REF

katjaweigel opened the pull request:
<!--
    Thank you for contributing to our project!

    Please do not delete this text completely, but read the text below and keep
    items that seem relevant. If in doubt, just keep everything and add your
    own text at the top, a reviewer will update the checklist for you.

    While the checklist is intended to be filled in by the technical and scientific
    reviewers, it is the responsibility of the author of the pull request to make
    sure all items on it are properly implemented.
-->

## Description
Time series, trends, and evaluation plots for regional (CORDEX) Historical changes in climate variables for REF
Based on existing monitoring and benchmarking recipes:
- recipe_portrait.yml
- monitor/recipe_monitor.yml
and 
- model_evaluation/recipe_model_benchmarking_timeseries.yml
- model_evaluation/recipe_model_benchmarking_boxplots.yml
from Benchmarking recipes (Lauer et al.), need PR #3598 to be merged

Trends TBD

So far using AR6 regions (from shape files). Will switch to CORDEX when shape file available.

<!--
    Please describe your changes here, especially focusing on why this pull request makes
    ESMValTool better and what problem it solves.

    Before you start, please read our contribution guidelines: https://docs.esmvaltool.org/en/latest/community/

    Please fill in the GitHub issue that is closed by this pull request, e.g. Closes #1903
-->
- Closes #3881
- Link to documentation: TBD

* * *

## Before you get started

<!--
    Please discuss your idea with the development team before getting started,
    to avoid disappointment or unnecessary work later. The way to do this is
    to open a new issue on GitHub.
-->

- [ x] [☝ Create an issue](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#contributing-code-and-documentation) to discuss what you are going to do

## Checklist

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

<!-- The next two lines turn the 🛠 and 🧪 below into hyperlinks -->
[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [ ] [🛠][1] This pull request has a [descriptive title](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-title)
- [ ] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#code-quality)
- [ ] [🛠][1] [Documentation](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#documentation) is available
- [ ] [🛠][1] [Tests](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#tests) run successfully
- [ ] [🛠][1] The [list of authors](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#list-of-authors) is up to date
- [ ] [🛠][1] Any changed dependencies have been [added or removed correctly](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#dependencies)
- [ ] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/en/latest/community/code_documentation.html#pull-request-checks) were successful

### [New or updated recipe/diagnostic](https://docs.esmvaltool.org/en/latest/community/diagnostic.html)

- [ ] [🧪][2] [Recipe runs successfully](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#testing-recipes)
- [ ] [🧪][2] [Recipe is well documented](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recipe-and-diagnostic-documentation)
- [ ] [🧪][2] [Figure(s) and data](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#diagnostic-output) look as expected from literature
- [ ] [🛠][1] [Provenance information](https://docs.esmvaltool.org/en/latest/community/diagnostic.html#recording-provenance) has been added

### [New or updated data reformatting script](https://docs.esmvaltool.org/en/latest/develop/dataset.html)

- [ ] [🛠][1] [Documentation](https://docs.esmvaltool.org/en/latest/community/dataset.html#dataset-documentation) is available
- [ ] [🛠][1] The dataset has been [added to the CMOR check recipe](https://docs.esmvaltool.org/en/latest/community/dataset.html#testing)
- [ ] [🛠][1] The dataset has been added to the shared [data pools](https://docs.esmvaltool.org/en/latest/community/dataset.html#cmorized-data) of DKRZ and Jasmin by the @ESMValGroup/OBS-maintainers team
- [ ] [🧪][2] Numbers and units of the data look [physically meaningful](https://docs.esmvaltool.org/en/latest/community/dataset.html#scientific-sanity-check)

***

To help with the number of pull requests:

-  🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValTool/pulls) in this repository

<!--
If you need help with any of the items on the checklists above, please do not hesitate to ask by commenting in the issue or pull request.
-->

katjaweigel commented:
Documentation is working now, but now the run test faild due to a reason, I don't understand (it was working on Friday and I only changed the documentation and merged the latest main this morning). I guess I'll merge a new main again later today ...

katjaweigel commented:
Does it make sense to use only some IPCC reference regions as example or should I actually add all? (Could be too many for some of the plots, one might to change the recipe to do several plots instead of one with more then 40 panels.)

schlunma commented on esmvaltool/diag_scripts/monitor/multi_datasets.py:
```diff
@@ -666,6 +670,10 @@
     from the corresponding dataset, e.g., ``{project}``, ``{short_name}``,
     ``{exp}``. Examples: ``title: 'Awesome Plot of {long_name}'``, ``xlabel:
     '{short_name}'``, ``xlim: [0, 5]``.
+title_metric: string, optional
+    Overwrites metric name in plot title. Optional for distance metrics.
+    Necessary, if a differnt metric is used (e.g. linear trend), because this
+    is else not identified and yieds arbitarry strings in the title.
```
```suggestion
```

This is not necessary. The same can be achieved with

```
pyplot_kwargs:
  subtitle: NEW TITLE
```

See my suggestion below

katjaweigel replied:
Unfortunately, 
```
pyplot_kwargs:
  suptitle: NEW TITLE
```
Has no effect for the benchmarking plot, the title remains "Near-Surface Air Temperatureof MIROC6", where "Near-Surface Air Temperature" is what results from `metric = cubes[0].long_name.partition("of")[0]` if there is no distance_metric.
(The result is the same, if I try to set suptitle as plot_kwargs, as figure_kwargs it results in an error.)
Setting an alternative for `metric = cubes[0].long_name.partition("of")[0]` was the easier way to fix that, but mybe it would be better, to make 
```
pyplot_kwargs:
  suptitle: NEW TITLE
```
work? 
(But I don't understand, why it is not working, since `self._process_pyplot_kwargs(plot_type, datasets[i])` is called in `def _plot_benchmarking_boxplot(self, df, cubes, variables, datasets):` )

katjaweigel replied:
In the current version I use the boxplot feature (with only two measurements, though
![benchmarking_boxplot_tas_MIROC6_Amon_historical_r1i1p1f1](https://github.com/user-attachments/assets/d9b3c33f-ece3-4241-92bd-b488a9b3dfbd)
)?

schlunma commented on esmvaltool/diag_scripts/monitor/multi_datasets.py:
```diff
@@ -647,6 +647,10 @@
     fontsize plus 2. Does not affect suptitles. If not given, use default
     matplotlib values. For a more fine-grained definition of fontsizes, use the
     option ``matplotlib_rc_params`` (see above).
+legend_kwargs: dict, optional
+    Optional keyword arguments for :func:`matplotlib.pyplot.legend`,
+    e.g. ``loc``. A legend is only shown if legend_kwargs are given,
+    default is no legend.
```
It looks like this is not used currently

katjaweigel replied:
Yes, it is currently not used anymore. The idea to include it, was to compare one model with the CMIP6 ensemble and observations/reanalysis. At the moment it was said not to include the CMIP6 ensemble for REF, therefore the recipe does not use it. But I'm not sure if I should remove it, since I think it is useful to include legends in some cases?

katjaweigel replied:
Removed

schlunma commented on esmvaltool/recipes/ref/recipe_model_benchmarking_boxplots_region_trend.yml:
```diff
@@ -0,0 +1,177 @@
+          figsize: [4, 5]
+          layout: constrained
+        # Overwrites metric in title, necessary, if no distance metric is used
+        title_metric: 'Linear trend (2001-2014)'
```
```suggestion
        pyplot_kwargs:
          suptitle: Linear of trend of MIROC for N.Europe
```

katjaweigel replied:
Does not effect the suptitle, see above?

schlunma reviewed (changes requested):
Thanks Katja! I had a brief first look at the diagnostic changes. Two comments:

1. As far as I can tell, there are only 2 real changes in the diagnostics, the addition of the options `title_metric` and `legend_kwargs` to the boxplots. It looks to me like these are not necessary (see my comments).
2. The other changes seem to come from the usage of an autoformatter. Please only use `pre-commit` (see https://github.com/ESMValGroup/ESMValTool/pull/3893 for details) for this, this should not introduce any changes to existing lines. Right now this is nearly impossible to review because so many lines are changed that you probably didn't touch.

Could you maybe try running the recipe with the published version of the diagnostic and check if the result is identical (using my comments on the recipe)? If yes, no changes on the diag are needed. You can checkout the old version by

```
cd ESMValTool
git fetch --all
git checkout origin/main -- esmvaltool/diag_scripts/monitor/multi_datasets.py
```

If you remote has another name than `origin` please adapt. Please also make sure to **commit or stash any uncommited changes you made to that file**, otherwise the command above will overwrite them.

hb326 commented on esmvaltool/recipes/ref/recipe_portrait_regions.yml:
```diff
@@ -0,0 +1,322 @@
+      shapefile: ar6
+      method: contains
+      ids: {Acronym: [WCE]}
+
```
I think we should have all the regions here, not just the five that are listed here. Since this is a super interesting diagnostic for the CORDEX community.

katjaweigel replied:
It is no problem to add more regions, but it will make the plot an the recipe huge. But I think it is important to first decide, what kind of portrait plot we want (see my answer to @bouweandela below)?

bouweandela commented on esmvaltool/recipes/ref/recipe_portrait_regions.yml:
```diff
@@ -0,0 +1,322 @@
+# ESMValTool
```
For the portrait plot produced by this recipe, would it be possible to store the reference dataset in a variable (e.g. with the same shape as the values) in the output NetCDF file?

The current output is:
```netcdf portrait {
dimensions:
	ppregion = 5 ;
	variable = 5 ;
	project = 1 ;
	split = 2 ;
variables:
	string ppregion(ppregion) ;
	string variable(variable) ;
	string project(project) ;
	string split(split) ;
	double var(ppregion, variable, project, split) ;
		var:_FillValue = NaN ;

// global attributes:
		:software = "Created with ESMValTool v2.13.0.dev16+gd3839331a" ;
		:caption = "RMSE performance metric" ;
data:

 ppregion = "ECA", "ENA", "MED", "SAH", "WCE" ;

 variable = "hus200", "pr", "psl", "tas", "ua200" ;

 project = "CMIP6" ;

 split = "Ref1", "Ref2" ;

 var =
  0, _,
  -0.26562990981659, 0,
  0.658703547559038, _,
  1.28437381550327, 1.11539806342596,
  -0.405895488144249, _,
  1.17306488773666, _,
  0.340033483139515, 0.114867407374071,
  0, _,
  -0.172380588347716, 0.0252619490659022,
  0.167941552698596, _,
  -0.0166655130349892, _,
  0, -0.112523783204075,
  -0.0904656465020044, _,
  0.0330545831008284, -0.08377315888959,
  0, _,
  1.57551232483152, _,
  -0.459794885038478, -0.753001370512609,
  -0.156387607108346, _,
  0, 0,
  -0.0389132823976344, _,
  -0.146790851371189, _,
  0.100536077681534, 0.0744045429479235,
  0.313707913215981, _,
  -0.201234689031456, -0.0893120691983471,
  0.199730102529329, _ ;
}
```
which makes it impossible to see what reference dataset was used. For the REF we are planning to generate our own portrait plots dynamically when users view the website,  based on user defined selection criteria. For example, they could select the variables, regions, or reference datasets they are interested in.

katjaweigel replied:
I'm not sure, if this should be the same as the portrait plot which is planned to be shown first for the REF?
As example for this portrait plot you linked in the last meeting this one: https://www.ilamb.org/CMIP6/historical/ 
It is structured data set (x) vs variable (y). Here it is region (x) vs variable (y). If you click on the plot you sent as example, you get plots for every variable for **all** data sets. I'm not sure, if it should remain like this, that there are **all** data sets in? (In the beginning, I guess only few will be available, so far I assumed, everything done for REF should work with one data set?)

If the idea is, that REF starts from the Data Set (x) vs Variable (y) portrait plot, we could either:
1) For any Data Set (x) vs Variable (y) show a plot for all regions (x) vs one variable (y = 1). [In this case we would assume, that all regions are available, which is probably true for global models and would fit to the comments from @hb326 to show all, not only a sample of regions]
2) Put several data sets (instead of several variables) on the y-axis (but then data sets would end up on the y-axis instead of on the x-axis) for single variables. 
3) Turning it around: data sets or one data set (x-axis) vs region (y-axis) for single variables (disadvantage: Grouping the IPCC regions into continents does not work [without modifications in the diagnostic])
4) Keep the current version regions(x) vs variables (y)
I think before writing out data it would be necessary to first agree on if we want one/several portrait plot(s) in there comparing the regions and how it should look like? 

For 1, 2 and 4 I see the disadvantage, that the plot gets very wide in x direction, if all regions are included?
For 3 I see the disadvantage, that a grouping of the IPCC regions into continents would not work at the moment [without modifications in the diagnostic]) but I could sort them alphabetic instead?
For 1 and 3 with one data set I see the disadvantage, that 1D portrait plots look strange, if there is only one row or column? (This could be replaced a type of plot which makes it easier to read the numbers, e.g. a bar plot)?
For 4 I see the disadvantage, that it doesn't fit as subplot to the global portrait plot split into variables? Therefore is really important, if that should be the case (meaning to split everything by variable) or not?

An additional option would be, to get several rows/columns by using several metrics (in addition to RMSE), but that would probably also not work out of the box (issues with the color bar)?

hb326 commented on esmvaltool/recipes/ref/recipe_model_benchmarking_timeseries_region.yml:
```diff
@@ -0,0 +1,342 @@
+        - operator: percentile
+          percent: 90
+      exclude: [reference_dataset, MIROC6]
+
```
The lines above we probably don't need if we do not show the CMIP6 model envelope, no?

hb326 commented on esmvaltool/recipes/ref/recipe_model_benchmarking_timeseries_region.yml:
```diff
@@ -0,0 +1,342 @@
+      tas: &var_default
+        start_year: 2001
+        end_year: 2014
+        preprocessor: pp
```
Can't we have a longer time period covered? I know that with the CMIP6 historical simulations we can only go up to 2014, but we could (and should) start earlier than 2001. HadCRUT5 should cover also earlier years, right?

katjaweigel replied:
Yes, I can add longer time series (but I would do that last since it increases calculation time). What would you prefer: As long as possible for every variable or a certain, eventually common period [the reanalysis/observations have different start dates, but I guess 1980 should work for most]? Which end date? Is there already one set for CMIP7 simulations?

hb326 replied:
I think the time period that you chose now is fine. We will discuss this a little more next week at the workshop, I think.

bouweandela commented:
> Does it make sense to use only some IPCC reference regions as example or should I actually add all? (Could be too many for some of the plots, one might to change the recipe to do several plots instead of one with more then 40 panels.)

A few examples is fine, the REF works by updating the existing recipes, e.g. the dataset or region, and then runs those.

hb326 commented on esmvaltool/recipes/ref/recipe_model_benchmarking_boxplots_region_trend.yml:
```diff
@@ -0,0 +1,177 @@
+  mip: Amon
+  exp: historical
+  ensemble: r1i1p1f1
+  start_year: 2001
```
Could we also have an earlier start date here? The historical simulations have the data, and the obs should have it as well, no?

katjaweigel replied:
Yes, but long time series I would include last (see also time series).

hb326 commented:
Thanks for the pull request, @katjaweigel!
I have tested all recipes, and they run through just fine. 
I think, though, the recipes would be good to go through some optimization (see comments).

katjaweigel commented on esmvaltool/recipes/ref/recipe_monitor_anncyc_regions.yml:
```diff
@@ -0,0 +1,49 @@
+      plot:
+        script: monitor/monitor.py
+        plots:
+          annual_cycle: {}
```
Unfortunately, I cannot easily do both at the moment. The diagnostics I found allow either to:
- make one (or few to keep them readable) multi panel plots for all IPCC regions (using monitor/monitor.py)
- compare the annual cycle to observations for one region (using monitor/multi_datasets.py like for the time series). So far, monitor/multi_datasets.py does not allow multi panel plots. I'm not sure if this is also going to change with the refactoring @schlunma works on?

The combination is as far as I know only possible with Irenes diagnostic ( regional_downscaling
/Figure9.40.ncl, it contains the outdated AR5 regions hard coded in the diagnostic) and I plan not to touch that again.

I guess what is more useful also depends on the structure, in which the results for the REF will be shown. If one will click (only) on the global portrait plot it would be probably useful to get a multi panel plot containing all regions. If the plan is to click on a regional portrait plot, it would probably be better to have the annual cycle for each region separately (compared to observations/reanalysis). This is the same issue for the time series, but I so far included different solutions (also to show what is there).

katjaweigel commented on esmvaltool/recipes/ref/recipe_model_benchmarking_boxplots_region_trend.yml:
```diff
@@ -0,0 +1,177 @@
+    convert_units:
+      units: mm kg m-3 day-1 year-1
+
+  trend_sst:
```
SST is one of the few variables with two reference data sets at the moment. Is it sure that we should only use the land regions (there are ocean regions as well)? The answer also depends on the discussion on how to show the trend at all from other comments ...

katjaweigel commented on esmvaltool/recipes/ref/recipe_model_benchmarking_boxplots_region_trend.yml:
```diff
@@ -0,0 +1,177 @@
+        plot_folder: '{plot_dir}'
+        group_variables_by: variable_group
+        plots:
+          benchmarking_boxplot:
```
There are a lot of different possibilities :)
I started with this one, because in the beginning a CMIP6 ensemble was in there as well. When we removed that and I asked if I should reduce the plot to boxes and scatter or only scatter you decided for the boxes (which is probably also because I use a layout, which was made for boxes). Unfortunately this, and that @schlunma doesn't really like changes in monitor/multi_datasets.py at the moment kind of makes me going back to the start again what to do with trends. There for I think, similar to the issue with the portrait plot, it is important to decide first, what should be shown. From the answers above I'm still not sure, if we:
- should show all (IPCC) regions (land: about 40; ocean?) together in one plot, eventually using multiple panels or better every region on its own?
- should have individual plots for every variable or combine several/all variables?
- make plots which can combine different data sets, work with a single data set or both?

Following the ILAMB plot  (https://www.ilamb.org/CMIP6/historical/) and the discussion about data availability I guess it would be better to make all plots for each variable individually (and not for a combination of variables as the one for trend are at the moment) despite eventually the portrait plot (see below)?  
Following @bouweandela answer about the regions above, dataset and region can be set individually.  But I'm not sure if that should be really the case for regions? I see different possibilities there. If we assume the starting point will be a plot like the one from ILAMB, when one clicks on one variable (and data set?) there we could either:
1) include a portrait plot similar to the one from ILAMB, now data set(s) (x-axis) vs region (y-axis). If one clicks on a region there you'd get again a new page with time series, annual cycle and trend for that one region (and one data set or several or possibly both?)
2) include the plots (portrait, time series, annual cycles, trends) on the same page for all regions (and one data set or several or possibly both?)

For the first case I would rather make individual plots for time series (available), annual cycle (can be done easily by modifying the recipe for time series) and trend (TBD) compared to reference data.
For the second case I would need multi panel plots for time series and annual cycle (easily available only without reference data at the moment) and trend (TBD, in this case I'd rather do a bar or scatter plot with all regions on the x-axis) and portrait plot data set(s) (x-axis) vs region (y-axis) or region (x-axis) vs data set(s) (y-axis) (should be easy to change the recipe). 

I guess I don't have the time to do both (and if there could be plots with several variables there are even more potential options)?

schlunma commented:
Thanks for all your work on this @katjaweigel! I just had a chat with @hb326 about this. To keep things as simple as possible, we suggest to

- remove `esmvaltool/recipes/ref/recipe_portrait_regions.yml` (this will probably be done by other metrics packages).
- change `esmvaltool/recipes/ref/recipe_monitor_anncyc_regions.yml` to include observational datasets (using `multi_datasets.py`) and create one plot per region (these will be combined by the REF later).
- leave `esmvaltool/recipes/ref/recipe_model_benchmarking_timeseries_region.yml` as is (with the extended time range, maybe just the final 30 years?).
- change `esmvaltool/recipes/ref/recipe_model_benchmarking_boxplots_region_trend.yml` so it shows simple bar plots instead of boxplots (see below).

Does that make sense?

Here is an example to create a simple barplot with the seaborn diagnostic:

![seaborn_barplot](https://github.com/user-attachments/assets/e46678ba-51b1-4153-83d4-37b79b71da40)

```yml
# ESMValTool
---
documentation:
  description: Test
  authors:
    - schlund_manuel
  title: Test.

datasets:
  - {dataset: BCC-CSM2-MR, project: CMIP6, exp: historical, ensemble: r1i1p1f1, grid: gn}
  - {dataset: MIROC6,      project: CMIP6, exp: historical, ensemble: r1i1p1f1, grid: gn}

preprocessors:
  trend:
    custom_order: true
    extract_shape:
      shapefile: ar6
      decomposed: true
      ids:
        Name:
          - N.Europe
          - Mediterranean
          - Equatorial.Pacific-Ocean
          - Central-Africa
          - E.C.Asia
          - New-Zealand
    area_statistics:
      operator: mean
    linear_trend:
    convert_units:
      units: K yr-1

diagnostics:
  tas_trends:
    variables:
      tas:
        preprocessor: trend
        mip: Amon
        timerange: P30Y/*
    scripts:
      plot:
        script: seaborn_diag.py
        seaborn_func: barplot
        seaborn_kwargs:
          x: shape_id
          y: tas
          hue: dataset
        add_aux_coords: true
        facets_as_columns: [dataset]
        legend_title: Model
        plot_object_methods:
          set_xlabel: region
          set_ylabel: tas trend [K/yr]
        seaborn_settings:
          style: ticks
          rc:
            figure.figsize: [14, 6]
        suptitle: Trend in Near-Surface Air Temperature

```

Due to an iris bug (will report soon) it is currently not possible to have one plot per region (scalar cubes cannot be converted to a pandas object).

katjaweigel commented on esmvaltool/recipes/ref/recipe_monitor_anncyc_regions.yml:
```diff
@@ -0,0 +1,49 @@
+# ESMValTool
```
I switched to another diagnostic, (the same as for time series) to include a reference data set. This should also provide provenance. However, it is strange, that this did not work for this one, because it was also only using an existing diagnostic (monitor/monitor.py used in monitor/recipe_monitor.yml ), which should provide provenance.

katjaweigel commented on esmvaltool/recipes/ref/recipe_model_benchmarking_timeseries_region.yml:
```diff
@@ -0,0 +1,342 @@
+# ESMValTool
```
I try to remember to remove it for all recipes. Not really sure, why it is there at all.

katjaweigel commented:
Thanks @schlunma for the overview and the recipe!
- I changed the recipes for time series and annual cycle now and used the same method as for time series to plot annual cycles for single regions.
- I removed esmvaltool/recipes/ref/recipe_portrait_regions.yml for now, but I still wonder if not better keep it?
- I made a new trend recipe based on the seaborn recipe you added above for all regions. If I unclude the layer from the 3D variables hus and ua (as for time series and annual cycle it is rather slow (more one hour now, still running but I have to leave now).  Additionally, it does not like "P30Y/*" for hus, but 1980/2014 works). So made a version with only tas, pr and psl at the moment, but it can also be, that it gets stuck on Levante for whatever reason, now the one with 3 Variables also does not get beyond the first one as fast as I had hoped.

![seaborn_barplot](https://github.com/user-attachments/assets/1df8dc3b-e5c8-4b5c-8494-d3f7bc6ada1b)

katjaweigel commented:
I had to turn off parallelization, then the 2D fields work. 3d still does not work due to memory issues (I guess I'd need to look into dask for that). At the moment the recipe therefore only contains the 2D fields.

schlunma commented on doc/sphinx/source/recipes/recipe_ref.rst:
```diff
@@ -13,16 +13,22 @@ Here ESMValTool recipes are collected which will be used in the CMIP
 -----------------
 
-Recipes are stored in recipes
+Recipes are stored in `recipes/ref`
```
```suggestion
Recipes are stored in `recipes/`
```

schlunma commented on doc/sphinx/source/recipes/recipe_ref.rst:
```diff
@@ -13,16 +13,22 @@ Here ESMValTool recipes are collected which will be used in the CMIP
 * :ref:`recipe_tcre.yml <recipes_tcre>`:
   Calculate transient climate response to cumulative CO2 emissions (TCRE)
-* ref/recipe_ref_cre.yml:
+* recipe_ref_cre.yml:
```
```suggestion
* ref/recipe_ref_cre.yml:
```

schlunma commented on doc/sphinx/source/recipes/recipe_ref.rst:
```diff
@@ -13,16 +13,22 @@ Here ESMValTool recipes are collected which will be used in the CMIP
+* ref/recipe_model_benchmarking_annual_cycle_region.yml:
+  Annual cycle in comparison with refernce data for a region defined through a shape file, based on :ref:`recipe_model_benchmarking_annual_cycle.yml <recipe_benchmarking>`
+* ref/recipe_seaborn_trend_regions.yml:
+  Linear Trends for all IPCC regions compaerd with refernce data, based on :ref:`recipe_seaborn.yml <recipes_seaborn_diag>`
```
```suggestion
  Linear Trends for all IPCC regions compared with reference data, based on :ref:`recipe_seaborn.yml <recipes_seaborn_diag>`
```

schlunma commented on doc/sphinx/source/recipes/recipe_ref.rst:
```diff
@@ -13,16 +13,22 @@ Here ESMValTool recipes are collected which will be used in the CMIP
+* recipe_ref_cre.yml:
   Maps and zonal means of longwave and shortwave cloud radiative effect
+* ref/recipe_model_benchmarking_timeseries_region.yml:
+  Time series in comparison with refernce data for a region defined through a shape file, based on :ref:`recipe_model_benchmarking_timeseries.yml <recipe_benchmarking>`
```
```suggestion
  Time series in comparison with reference data for a region defined through a shape file, based on :ref:`recipe_model_benchmarking_timeseries.yml <recipe_benchmarking>`
```

schlunma commented on doc/sphinx/source/recipes/recipe_ref.rst:
```diff
@@ -13,16 +13,22 @@ Here ESMValTool recipes are collected which will be used in the CMIP
+* ref/recipe_model_benchmarking_timeseries_region.yml:
+  Time series in comparison with refernce data for a region defined through a shape file, based on :ref:`recipe_model_benchmarking_timeseries.yml <recipe_benchmarking>`
+* ref/recipe_model_benchmarking_annual_cycle_region.yml:
+  Annual cycle in comparison with refernce data for a region defined through a shape file, based on :ref:`recipe_model_benchmarking_annual_cycle.yml <recipe_benchmarking>`
```
```suggestion
  Annual cycle in comparison with reference data for a region defined through a shape file, based on :ref:`recipe_model_benchmarking_annual_cycle.yml <recipe_benchmarking>`
```

schlunma commented on esmvaltool/recipes/ref/recipe_seaborn_trend_regions.yml:
```diff
@@ -0,0 +1,182 @@
+          - 'NCA'
+          - 'SCA'
+          - 'CAR'
+          - 'NWS'
+          - 'NSA'
+          - 'NES'
+          - 'SAM'
+          - 'SWS'
+          - 'SES'
+          - 'SSA'
+          - 'NEU'
+          - 'WCE'
+          - 'EEU'
+          - 'MED'
+          - 'SAH'
+          - 'WAF'
+          - 'CAF'
+          - 'NEAF'
+          - 'SEAF'
+          - 'WSAF'
+          - 'ESAF'
+          - 'MDG'
+          - 'RAR'
+          - 'WSB'
+          - 'ESB'
+          - 'RFE'
+          - 'WCA'
+          - 'ECA'
+          - 'TIB'
+          - 'EAS'
+          - 'ARP'
+          - 'SAS'
+          - 'SEA'
+          - 'NAU'
+          - 'CAU'
+          - 'EAU'
+          - 'SAU'
+          - 'NZ'
+          - 'WAN'
+          - 'EAN'
```
```suggestion
```

If these are all regions you can simply omit `ids`.

katjaweigel replied:
At the moment these are all land regions (46), if I use:
```
    extract_shape:
      shapefile: ar6
      decomposed: true
      ids:
        Acronym:
```
I get the 15 ocean regions in addition (making the plot even larger and leading to memory issues on Levante even for the 2D variables). (Without (ids: Acronym:) I get the names which are too long to write them on the x-axis ticks.) Not sure, what the best choice is:
- only land (current version)
- land and ocean in one plot or separated
- any other separation or reduction of regions
?

schlunma replied:
Ah, in this case just leave it like this. We let the scientific reviewer decide about the regions 😉 Maybe add a tiny comment that these are all the land regions?

katjaweigel replied:
Ok, thanks! I added that in the description. And I found the issue with the 3D variables now, I actually forgot to extract the 200hPa layer. Including this, the memory issue is also gone (above that was also caused by a 3D variable, I just did not realize that, sorry)

schlunma reviewed (approved):
Thanks @katjaweigel, code looks great! Just have some minor comments.

This is ready for a scientific review.

Note: I did not actually run the recipes.

katjaweigel commented:
@schlunma thanks a lot for checking! I included most of the changes you suggested, only for the regions I'm not sure what to do best (see above)?

hb326 commented:
@katjaweigel: I just ran the recipes for the annual cycles and the time series. Both ran fine. 👍 
I am testing now the trends, and I ran into some memory issues as well. I think the easiest solution would be to just provide the trend for ONE region at the time (like it is done for the other recipes), not all the regions at the same time. 
I also saw that you have the color red for the model line hardcoded as "MIROC6" right now in the recipes. I think it would be better to have that line definition open for all the models, so I would set it as default. The specific definitions for the obs are fine.

hb326 commented on doc/sphinx/source/recipes/recipe_ref.rst:
```diff
@@ -23,6 +23,12 @@ Recipes are stored in recipes
+* ref/recipe_model_benchmarking_annual_cycle_region.yml:
+  Annual cycle in comparison with reference data for a region defined through a shape file, based on :ref:`recipe_model_benchmarking_annual_cycle.yml <recipe_benchmarking>`
+* ref/recipe_seaborn_trend_regions.yml:
+  Linear Trends for all IPCC regions compared with reference data, based on :ref:`recipe_seaborn.yml <recipes_seaborn_diag>`
```
Here I would just use one region, as you do for the other two recipes. Then you can adjust the text to be as general as the texts in the description of the other recipes before.

hb326 commented on doc/sphinx/source/recipes/recipe_ref.rst:
```diff
@@ -82,3 +88,11 @@ Example plots:
+   :width:   8cm
+
+   Time series of near-surface air temperature anomalies from MIROC6 compared with HadCRUT5
+   for N.Europe for the years 1980 to 2014 (refernce period 1980 to 2009). Created with recipe_model_benchmarking_timeseries_region.yml.
```
There is a typo in the word "reference".

katjaweigel replied:
Thanks, I removed it

hb326 commented on esmvaltool/recipes/ref/recipe_model_benchmarking_annual_cycle_region.yml:
```diff
@@ -0,0 +1,149 @@
+          annual_cycle:
+            annual_mean_kwargs: False
+            plot_kwargs:
+              MIROC6:
```
I think it would be better if you use the "default" here for the models. Then all the models will have the same red line, and we do not have to hardcode the model name in the recipe for all the times when the recipe is run for the REF.

katjaweigel replied:
Thanks, changed

hb326 commented on esmvaltool/recipes/ref/recipe_model_benchmarking_timeseries_region.yml:
```diff
@@ -0,0 +1,295 @@
+          timeseries:
+            annual_mean_kwargs: False
+            plot_kwargs:
+              MIROC6:
```
Again here, it would be great if we could have the model lines as "default".

katjaweigel replied:
Thanks, changed

hb326 commented on esmvaltool/recipes/ref/recipe_model_benchmarking_timeseries_region.yml:
```diff
@@ -0,0 +1,295 @@
+                linestyle: '-'
+                linewidth: 2
+                zorder: 3
+              MultiModelPercentile10:
```
You can probable remove the lines 190 to 201 since we do not work with the envelopes in this recipe.

hb326 commented on esmvaltool/recipes/ref/recipe_model_benchmarking_timeseries_region.yml:
```diff
@@ -0,0 +1,295 @@
+          timeseries:
+            annual_mean_kwargs: False
+            plot_kwargs:
+              MIROC6:
```
Also here, set the models as "default".

katjaweigel replied:
Thanks, changed

hb326 commented on esmvaltool/recipes/ref/recipe_seaborn_trend_regions.yml:
```diff
@@ -0,0 +1,309 @@
+      shapefile: ar6
+      decomposed: true
+      ids:
+        Acronym:
```
I think it would be better to have the trends available for each individual region, not just as a collection of all regions. Most likely the results will be assembled in a new way (e.g. time series, annual cycle and trends per model per region) from the REF anyway, and then it might be better to just have the trend bar plot available for each individual region.

katjaweigel replied:
I'm not sure if that is really the case, but for consistency I can do it for single regions. Unfortunately, there is an iris bug causing issues with pandas (as @schlunma mentioned above), so we cannot simply plot only one region in the same style as many at the moment. But I managed to find a recipe setting, which allows it in anyway (using dataset both as x and hue). With this, the plots would look like this:
![seaborn_barplot](https://github.com/user-attachments/assets/fcd8b004-a308-4354-bee1-7fecd4b967ae)

Would you prefer this version?

hb326 replied:
That would be great! I know for sure that the REF will be doing all kinds of additional grouping and presentation of the results. So having the trends in for one model and one region in one bar plot would be great!

katjaweigel replied:
Unfortunately it seems not to be possible. Only for the 3D variables I managed to set one region but different air-pressure levels:
![seaborn_barplot](https://github.com/user-attachments/assets/8b540b12-62b4-4446-9abe-d73b78be297d)

(but at leas for hus that is maybe not the best choice because of its huge vertical gradient)
For the 2D ones I did not find any other split within a region, which works so far (I'd hoped for zonal_statistics, but it has an issue with NaNs)

katjaweigel replied:
@hb326 if we really need one model + one variable + one region now, I guess I could try to switch to a different diagnostic, there are some more which do bar plots (e.g. one of the deangelis ones). But I doubt that I would manage without changing a diagnositc.

katjaweigel replied:
Thanks @hb326 ! So I left the trend recipe now as it is and changed the other ones following your comments.

katjaweigel replied:
I made an error with double default, update soon

katjaweigel replied:
fixed :)

hb326 commented on esmvaltool/recipes/ref/recipe_seaborn_trend_regions.yml:
```diff
@@ -0,0 +1,309 @@
+        suptitle: Trend in Precipitation
+
+
+  psl_trends:
```
Does it make sense to show the trends for psl? Might be difficult to interpret?

katjaweigel replied:
I put the trend for the same variables as the time series and annual cycles. I think for regional analysis it is ok, to look at psl? But I could also skip it/use other variables if you prefere?

hb326 replied:
@axel-lauer and I had a discussion about the plot, and it seemed a little hard to interpret for the different regions. But for consistency with the other recipes, I would say we leave it it for now, and then see what kind of feedback we get for the REF diagnostics.

hb326 commented:
I ran the recipe "recipe_seaborn_trend_regions.yml" now and it ran through, after I allocated a whole node for it.

hb326 commented:
I reran the recipes again after I renamed them (to be more consistent with the other REF recipes). I updated the documentation accordingly.
All ran through just fine. Thanks for all your work on this, @katjaweigel!
This is ready to me merged.

hb326 reviewed (approved):
Ready to me merged!

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_trend_regions.yml:
```diff
@@ -0,0 +1,309 @@
+        mip: Amon
+        timerange: 1980/2009
+        additional_datasets:
+          - {dataset: ERA5, project: native6, type: reanaly, version: v1,
```
Would it be possible to use the data from obs4MIPs instead here?

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_trend_regions.yml:
```diff
@@ -0,0 +1,309 @@
+        mip: Amon
+        timerange: 1980/2009
+        additional_datasets:
+          - {dataset: ERA5, project: native6, type: reanaly, version: v1,
```
Would it be possible to use the data from obs4MIPs instead here?

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_trend_regions.yml:
```diff
@@ -0,0 +1,309 @@
+        additional_datasets:
+          - {dataset: HadCRUT5, project: OBS, type: ground,
+             version: 5.0.1.0-analysis, tier: 2}
+          - {dataset: ERA5, project: native6, type: reanaly, version: v1,
```
Would it be possible to use the data from obs4MIPs instead here?

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_timeseries_region.yml:
```diff
@@ -0,0 +1,271 @@
+        <<: *var_default_abs
+        reference_dataset: ERA5
+        additional_datasets:
+          - {dataset: ERA5, project: native6, type: reanaly, version: v1,
```
Would it be possible to use the data from obs4MIPs instead here?

bouweandela commented:
Could you please check if observational data is already available in obs4MIPs? https://github.com/ESMValGroup/ESMValTool/issues/3932#issuecomment-2815333478 For example, for ERA-5 obs4MIPs provides the following variables: 'psl', 'ta', 'tas', 'ua', 'uas', 'va', 'vas', 'zg' and GPCP-SG version 2.3 is available as "GPCP-V2.3".

katjaweigel commented:
> Could you please check if observational data is already available in obs4MIPs? [#3932 (comment)](https://github.com/ESMValGroup/ESMValTool/issues/3932#issuecomment-2815333478) For example, for ERA-5 obs4MIPs provides the following variables: 'psl', 'ta', 'tas', 'ua', 'uas', 'va', 'vas', 'zg' and GPCP-SG version 2.3 is available as "GPCP-V2.3".

Do we have them on Levante @hb326 ? The only one I find in Tier1 is:
`/work/bd0854/DATA/ESMValTool2/OBS/Tier1/GPCP-SG/pr_GPCP-SG_L3_v2.3_197901-201710.nc`
(no other variables, no ERA5)
Are there obs4MIPS data in a different folder or should I download them or is somebody already doing that?

katjaweigel commented on esmvaltool/recipes/ref/recipe_ref_timeseries_region.yml:
```diff
@@ -0,0 +1,271 @@
+        <<: *var_default_abs
+        reference_dataset: GPCP-SG
+        additional_datasets:
+          - {dataset: GPCP-SG, project: OBS, type: atmos, version: 2.3, tier: 2,
```
Changed

katjaweigel commented:
> Unfortunately, the only server providing that data is offline..

It is running again, I try to get them now, but I won't manage to update and test the recipes today anymore.

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_timeseries_region.yml:
```diff
@@ -0,0 +1,283 @@
+        <<: *var_default
+        reference_dataset: GPCP-SG
+        additional_datasets:
+          - {dataset: GPCP-SG, project: obs4MIPs, type: atmos, version: 2.3, tier: 1,
```
```suggestion
          - {dataset: GPCP-V2.3, project: obs4MIPs, tier: 1,
```

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_timeseries_region.yml:
```diff
@@ -0,0 +1,283 @@
+        <<: *var_default
+        reference_dataset: ERA-5
+        additional_datasets:
+          - {dataset: ERA-5, project: obs4MIPs, type: atmos, version: PCMDI, tier: 1,
```
```suggestion
          - {dataset: ERA-5, project: obs4MIPs, tier: 1,
```

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_timeseries_region.yml:
```diff
@@ -0,0 +1,283 @@
+        reference_dataset: ERA-5
+        preprocessor: pp_200
+        additional_datasets:
+          - {dataset: ERA-5, project: obs4MIPs, type: atmos, version: PCMDI, tier: 1,
```
```suggestion
          - {dataset: ERA-5, project: obs4MIPs, tier: 1,
```

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_timeseries_region.yml:
```diff
@@ -0,0 +1,283 @@
+        <<: *var_default_abs
+        reference_dataset: GPCP-SG
+        additional_datasets:
+          - {dataset: GPCP-SG, project: obs4MIPs, type: atmos, version: 2.3, tier: 1,
```
```suggestion
          - {dataset: GPCP-V2.3, project: obs4MIPs, tier: 1,
```

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_timeseries_region.yml:
```diff
@@ -0,0 +1,283 @@
+        <<: *var_default_abs
+        reference_dataset: ERA-5
+        additional_datasets:
+          - {dataset: ERA-5, project: obs4MIPs, type: atmos, version: PCMDI, tier: 1,
```
```suggestion
          - {dataset: ERA-5, project: obs4MIPs, tier: 1,
```

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_timeseries_region.yml:
```diff
@@ -0,0 +1,283 @@
+        reference_dataset: ERA-5
+        preprocessor: pp_abs_200
+        additional_datasets:
+          - {dataset: ERA-5, project: obs4MIPs, type: atmos, version: PCMDI, tier: 1,
```
```suggestion
          - {dataset: ERA-5, project: obs4MIPs, tier: 1,
```

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_annual_cycle_region.yml:
```diff
@@ -0,0 +1,149 @@
+        <<: *var_default_abs
+        reference_dataset: ERA-5
+        additional_datasets:
+          - {dataset: ERA-5, project: obs4MIPs, type: atmos, version: PCMDI, tier: 1,
```
```suggestion
          - {dataset: ERA-5, project: obs4MIPs, tier: 1,
```
`type` is not needed for obs4MIPs data and the only available version is `v20250220`, but the version does not need to be specified because ESMValCore will automatically use the latest available version.

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_annual_cycle_region.yml:
```diff
@@ -0,0 +1,149 @@
+        reference_dataset: ERA-5
+        preprocessor: pp_abs_200
+        additional_datasets:
+          - {dataset: ERA-5, project: obs4MIPs, type: atmos, version: PCMDI, tier: 1,
```
```suggestion
          - {dataset: ERA-5, project: obs4MIPs, tier: 1,
```

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_annual_cycle_region.yml:
```diff
@@ -0,0 +1,149 @@
+        <<: *var_default_abs
+        reference_dataset: GPCP-SG
+        additional_datasets:
+          - {dataset: GPCP-SG, project: obs4MIPs, type: atmos, version: 2.3, tier: 1,
```
```suggestion
          - {dataset: GPCP-V2.3, project: obs4MIPs, tier: 1,
```
The name of the obs4MIPs dataset on ESGF is `GPCP-V2.3`. `type` is not a required facet for obs4MIPs data. The only available version on ESGF is `v20180519`, but the version does not need to be specified because ESMValCore will automatically use the latest available version.

bouweandela replied:
You are right that `GPCP-SG` is used in the filename, but what matters is the value for the `source_id`, as that is what we call `dataset` in ESMValCore (see [here](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/api/esmvalcore.esgf.html#esmvalcore.esgf.facets.FACETS) for the mapping from 'our' facet names to ESGF names). [Using `source_id=GPCP-V2.3` in an ESGF search](https://esg-dn1.nsc.liu.se/esg-search/search?format=application%2Fsolr%2Bjson&limit=500&distrib=true&offset=0&type=File&project=obs4MIPs&source_id=GPCP-V2.3) returns a file:
```json
{
  "responseHeader":{
    "status":0,
    "QTime":5,
    "params":{
      "df":"text",
      "q.alt":"*:*",
      "indent":"true",
      "echoParams":"all",
      "fl":"*,score",
      "start":"0",
      "fq":["type:File",
        "project:\"obs4MIPs\"",
        "source_id:\"GPCP-V2.3\""],
      "sort":"id asc",
      "rows":"500",
      "q":"*:*",
      "shards":"localhost/solr/files,localhost:8985/solr/files,localhost:8986/solr/files,localhost:8987/solr/files,localhost:8988/solr/files,localhost:8989/solr/files,localhost:8990/solr/files",
      "tie":"0.01",
      "facet.limit":"-1",
      "qf":"text",
      "facet.method":"enum",
      "facet.mincount":"1",
      "wt":"json",
      "facet":"true",
      "facet.sort":"lex"}},
  "response":{"numFound":1,"start":0,"maxScore":1.0,"docs":[
      {
        "id":"obs4MIPs.NASA-GSFC.GPCP.atmos.mon.v20180519.pr_GPCP-SG_L3_v2.3_197901-201710.nc|dpesgf03.nccs.nasa.gov",
        "version":"1",
        "cf_standard_name":["precipitation_flux"],
        "checksum":["4dd4678b79ef139446c8406da5aae4fed210abb2f2160ef95f6988bf83e4525b"],
        "checksum_type":["SHA256"],
        "data_node":"dpesgf03.nccs.nasa.gov",
        "dataset_id":"obs4MIPs.NASA-GSFC.GPCP-V2.3.atmos.mon.v20180519|dpesgf03.nccs.nasa.gov",
        "dataset_id_template_":["%(project)s.%(institute)s.%(source_id)s.%(realm)s.%(time_frequency)s"],
        "index_node":"esgf-node.llnl.gov",
        "instance_id":"obs4MIPs.NASA-GSFC.GPCP.atmos.mon.v20180519.pr_GPCP-SG_L3_v2.3_197901-201710.nc",
        "institute":["NASA-GSFC"],
        "latest":true,
        "master_id":"obs4MIPs.NASA-GSFC.GPCP.atmos.mon.pr_GPCP-SG_L3_v2.3_197901-201710.nc",
        "model":["Obs-GPCP"],
        "product":["observations"],
        "project":["obs4MIPs"],
        "realm":["atmos"],
        "replica":false,
        "size":19348352,
        "source_id":["GPCP-V2.3"],
        "time_frequency":["mon"],
        "timestamp":"2018-02-17T17:17:54Z",
        "title":"pr_GPCP-SG_L3_v2.3_197901-201710.nc",
        "tracking_id":["4070c751-6c2d-440f-a4d7-5b325fb98990"],
        "type":"File",
        "url":["https://dpesgf03.nccs.nasa.gov/thredds/fileServer/obs4MIPs/observations/NASA-GSFC/Obs-GPCP/GPCP/V2.3/atmos/pr/pr_GPCP-SG_L3_v2.3_197901-201710.nc|application/netcdf|HTTPServer"],
        "variable":["pr"],
        "variable_long_name":["Precipitation"],
        "variable_units":["kg m-2 s-1"],
        "_version_":1662343940370595840,
        "retracted":false,
        "_timestamp":"2020-03-27T18:45:20.965Z",
        "score":1.0}]
  },
  "facet_counts":{
    "facet_queries":{},
    "facet_fields":{},
    "facet_ranges":{},
    "facet_intervals":{},
    "facet_heatmaps":{}}}
```
while a search [using `source_id=GPCP-SG` in an ESGF search](https://esg-dn1.nsc.liu.se/esg-search/search?format=application%2Fsolr%2Bjson&limit=500&distrib=true&offset=0&type=File&project=obs4MIPs&source_id=GPCP-SG) returns no files:
```json
{
  "responseHeader":{
    "status":0,
    "QTime":8,
    "params":{
      "df":"text",
      "q.alt":"*:*",
      "indent":"true",
      "echoParams":"all",
      "fl":"*,score",
      "start":"0",
      "fq":["type:File",
        "project:\"obs4MIPs\"",
        "source_id:\"GPCP-SG\""],
      "sort":"id asc",
      "rows":"500",
      "q":"*:*",
      "shards":"localhost/solr/files,localhost:8985/solr/files,localhost:8986/solr/files,localhost:8987/solr/files,localhost:8988/solr/files,localhost:8989/solr/files,localhost:8990/solr/files",
      "tie":"0.01",
      "facet.limit":"-1",
      "qf":"text",
      "facet.method":"enum",
      "facet.mincount":"1",
      "wt":"json",
      "facet":"true",
      "facet.sort":"lex"}},
  "response":{"numFound":0,"start":0,"maxScore":0.0,"docs":[]
  },
  "facet_counts":{
    "facet_queries":{},
    "facet_fields":{},
    "facet_ranges":{},
    "facet_intervals":{},
    "facet_heatmaps":{}}}
```

katjaweigel replied:
But when I write "dataset: GPCP-V2.3" in the recipe it does not run (maybe it would, if I'd allow download?) but it does with dataset: GPCP-SG (because on Levante, these data are available)?

katjaweigel replied:
Thanks for the answer @bouweandela ! I tried again with `search_esgf: when_missing` in config, then it works with "GPCP-V2.3". But it does not work any more, when I set it to ` search_esgf: never ` to run the next recipe, which needs the same data (or even to rerun the same recipe again). Then it still does not find the data? The reason seem to be, that it does not create a folder. It puts the data under:
(esmvaltool) [b380216@levante1 ESMValTool]$ ls -lhrt /work/bd1083/b380216/extraobsv2/Tier1/
total 19M
drwxr-sr-x 2 b380216 bd0854 20K Jun 15 12:28 ERA-5
-rwx------ 1 b380216 bd0854 19M Jun 25 15:39 pr_GPCP-SG_L3_v2.3_197901-201710.nc

When I move the data by hand into a sub-folder called "GPCP-V2.3" it works, so I change it in these recipes now to GPCP-V2.3 now.
But I wonder if we need to change something for the downloader in cases where file name and the source_id disagree
and/or if we then should rename?
/work/bd0854/DATA/ESMValTool2/OBS/Tier1/GPCP-SG
to
/work/bd0854/DATA/ESMValTool2/OBS/Tier1/GPCP-V2.3
? 
But this would cause other recipes, which currently use "GPCP-SG" not to work anymore on Levante?
These recipes are:
- esmvaltool/recipes/recipe_model_benchmarking_timeseries_pr_e.yml
- esmvaltool/recipes/examples/recipe_check_obs.yml
- esmvaltool/recipes/ipccwg1ar5ch9/recipe_flato13ipcc_figure_96.yml
- esmvaltool/recipes/ipccwg1ar6ch3/recipe_ipccwg1ar6ch3_atmosphere.yml
- esmvaltool/recipes/model_evaluation/recipe_model_benchmarking_boxplots.yml
- esmvaltool/recipes/model_evaluation/recipe_model_benchmarking_maps.yml
- esmvaltool/recipes/model_evaluation/recipe_model_evaluation_basics.yml
- esmvaltool/recipes/model_evaluation/recipe_model_evaluation_clouds_clim.yml
- esmvaltool/recipes/model_evaluation/recipe_model_evaluation_precip_zonal.yml
-esmvaltool/recipes/ref/recipe_ref_scatterplot.yml

There is also still the issue, that I cannot find pr and hus as cmorized ERA5 data?

bouweandela replied:
> But it does not work any more, when I set it to `search_esgf: never` to run the next recipe, which needs the same data (or even to rerun the same recipe again). 

You need to set the rootpath and DRS so it matches the directory tree in your config-user.yml file. For example, mine looks like this for the obs4MIPs project on Levante:
```yaml
rootpath:
  obs4MIPs:
    /work/bd0854/DATA/ESMValTool2/OBS: default
    /work/bd0854/DATA/ESMValTool2/download: ESGF
```
> I wonder if we need to change something for the downloader

The downloader uses DRS `ESGF`, for obs4MIPs that is https://github.com/ESMValGroup/ESMValCore/blob/4d8fcbfb331e0745db58321cc41fe7de88c82393/esmvalcore/config-developer.yml#L110-L119
to store files. As you can see, the only requirement on the filename is that it starts with the `short_name` for that DRS setting, so what comes after it should't matter.

katjaweigel replied:
There is no issue with the file name, but with the name of the directory, which is usually the data set name:
It downloads the data to:
Tier{tier}/
but expects: 'Tier{tier}/{dataset}' for finding the data, if they are not downloaded in the same run.
I'd expect that most users don't change config_developer (because it is part of ESMValCore) so if I use `dataset: GPCP-V2.3` it expects this as directory name and won't find the data if they are directly under Tier{tier}/ or in `/work/bd0854/DATA/ESMValTool2/OBS/Tier1/GPCP-SG` ?

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_trend_regions.yml:
```diff
@@ -0,0 +1,309 @@
+        mip: Amon
+        timerange: 1980/2009
+        additional_datasets:
+          - {dataset: GPCP-SG, project: obs4MIPs, type: atmos, version: 2.3, tier: 1,
```
```suggestion
          - {dataset: GPCP-V2.3, project: obs4MIPs, tier: 1,

```

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_trend_regions.yml:
```diff
@@ -0,0 +1,309 @@
+        additional_datasets:
+          - {dataset: GPCP-SG, project: obs4MIPs, type: atmos, version: 2.3, tier: 1,
+             reference_for_metric: true}
+          - {dataset: ERA5, project: native6, type: reanaly, version: v1,
+             tier: 3, reference_for_metric: true}
```
```suggestion
          - {dataset: ERA-5, project: obs4MIPs, tier: 1, reference_for_metric: true}
```

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_trend_regions.yml:
```diff
@@ -0,0 +1,309 @@
+        mip: Amon
+        timerange: 1980/2009
+        additional_datasets:
+          - {dataset: ERA-5, project: obs4MIPs, type: atmos, version: PCMDI, tier: 1,
```
```suggestion
          - {dataset: ERA-5, project: obs4MIPs, tier: 1,

```

bouweandela commented on esmvaltool/recipes/ref/recipe_ref_trend_regions.yml:
```diff
@@ -0,0 +1,309 @@
+        mip: Amon
+        timerange: 1980/2009
+        additional_datasets:
+          - {dataset: ERA-5, project: obs4MIPs, type: atmos, version: PCMDI, tier: 1,
```
```suggestion
          - {dataset: ERA-5, project: obs4MIPs, tier: 1,

```
